### ISSUE TITLE
[data grid] align column data and header with type "number"

### ISSUE THREAD

**Issue by hlebshypulahub** [mui/mui-x#3291]
### Current behavior 😯

When I set the type of a column to type: "number" the column header and data align to the right.

### Expected behavior 🤔

I want the column of type: "number" align to the left like other columns with default type.

### Steps to reproduce 🕹

I reproduced the issue at the simple example from documentation: [codesandbox](https://codesandbox.io/s/exciting-leftpad-f7r2v?file=/demo.js)
The third column age has type: "number" and everything alignes to the right. But other columns with default types - to the left.

**Comment by oliviertassinari** [comment]
This seems to be what @hlebshypulahub was looking for: the `align` and `headerAlign` prop.

```jsx
import * as React from "react";
import { DataGrid } from "@mui/x-data-grid";

const rows = [
  {
    id: 1,
    username: "@MUI",
    age: 20
  }
];

export default function ColumnFluidWidthGrid() {
  return (
    <div style={{ height: 250, width: "100%" }}>
      <DataGrid
        columns={[
          {
            field: "id",
            flex: 1,
            minWidth: 150
          },
          {
            field: "username",
            width: 200
          },
          {
            field: "age",
            flex: 0.3,
            minWidth: 50,
            type: "number",
            align: "left",
            headerAlign: "left"
          }
        ]}
        rows={rows}
      />
    </div>
  );
}
```

https://codesandbox.io/s/black-cherry-gmiw9?file=/demo.js:0-714

Now, I can imagine why it wasn't clear. We might have a couple of problems in the docs:

1. https://mui.com/api/data-grid/grid-col-def/#properties We might want to cross-reference `headerAlign` and `align` in the description of the props. When you modify one you also likely want to modify the other. Initially, I got confused on why `align` was only modifying the cells and not the header. 
2. The type of `GridAlignment` which says nothing what are the accepted values
3. Maybe we should mention this in https://mui.com/components/data-grid/columns/

**Comment by QuinnrElder** [comment]
Afternoon!

I wanted to reach out today to see if there was a solution to this issue outside the use of ```align & headerAlign```? While using the DataGrid and applying the prop of ``` type: 'number'``` by default the flex-direction changes and CSS classes are added to the element. (found this using in the dev tools) Is this expected behavior and if so the docs are not clear as to why this is happening. If not, why does this happen only when applying the prop of ``` type: 'number'```?

Thank you!

**Comment by therour** [comment]
Hello @oliviertassinari 

current behavior:
If we added `align` and `headerAlign` to `type: 'number'`, found out on DataGrid, the menuIcon on the header (three dots) still on the left, 

Im expecting, if we add `headerAlign: 'left'` the menuIcon will be moved to the right, but it isnt...

I think the column definition should be having `menuIconAlign` too..

**Comment by aaarichter** [comment]
> https://codesandbox.io/s/black-cherry-gmiw9?file=/demo.js:0-714

that worked for me - but it is not clear why this type aligns differently than others

**Comment by LumaKernel** [comment]
> that worked for me - but it is not clear why this type aligns differently than others

Agree. I'm talking with my team's designer, and possibly can decide to change text-align. But I have concern that I missed some consideration...?

**Comment by LumaKernel** [comment]
not only numbers, but also boolean is centered.

header align looks defaulted to cell align

**Comment by LumaKernel** [comment]
Now I'm believing it's just about design decision.

**Comment by LumaKernel** [comment]
If I had to raise the issue, when we widen number columns much, header and cell value will become far apart.

**Comment by michelengelen** [comment]
`align` and `headerAlign` are the props for this, as shown above, and this also resolves the follow-up question about the menu icon: setting `headerAlign: 'left'` puts the header back in the unstyled default layout (title on the left, menu icon on the right), the same as any column without an explicit `headerAlign`. Only `center` and `right` need special-casing to reposition the icon, and both are already handled.

Closing as completed. The right-alignment on `number`/`boolean` types by default remains an intentional design choice (values of these types read better right/center-aligned), not a bug.
